### PR TITLE
Skill and hero specialty bonus description

### PR DESCRIPTION
**Pull request by godric3**
Fix for https://bugs.vcmi.eu/view.php?id=3011

### CODE REVIEW

File: lib/CHeroHandler.h [hunk 1]
```diff
@@ -135,7 +135,7 @@ class DLL_LINKAGE CHero
 
 // convert deprecated format
 std::vector<std::shared_ptr<Bonus>> SpecialtyInfoToBonuses(const SSpecialtyInfo & spec, int sid);
-std::vector<std::shared_ptr<Bonus>> SpecialtyBonusToBonuses(const SSpecialtyBonus & spec);
+std::vector<std::shared_ptr<Bonus>> SpecialtyBonusToBonuses(const SSpecialtyBonus & spec, int sid);
```
Comment (DjWarmonger) [inline lib/CHeroHandler.h]: How about default sid value to make this argument optional?

Comment (dydzio0614) [reply lib/CHeroHandler.h]: optional argument... maybe boost::optional will be useful for this?

Comment (DjWarmonger) [reply lib/CHeroHandler.h]: Nope, I meant just default `sid = 0` so coder doesn't need to write anything to skip this new feature.

Comment (godric3) [reply lib/CHeroHandler.h]: I could add it, but is there reason to? This function is used for converting deprecated format, so it's not likely to be used more

**Pull request merged**